mango-core/src/main/scala/org/bdgenomics/mango/models/AlignmentRecordMaterialization.scala

```diff
@@ -196,56 +135,58 @@ object AlignmentRecordMaterialization extends Logging {
+        AlignmentRecordField.start, AlignmentRecordField.readMapped, AlignmentRecordField.readGroupId,
+        AlignmentRecordField.end, AlignmentRecordField.sequence, AlignmentRecordField.cigar, AlignmentRecordField.readNegativeStrand,
+        AlignmentRecordField.readPaired, AlignmentRecordField.readGroupSampleId)
+
+      val alignmentRecordDataset: AlignmentRecordDataset =
+        if (regions.isDefined) {
+          if (sc.isPartitioned(fp)) {
+
+            // finalRegions includes references both with and without "chr" prefix
+            val finalRegions: Iterable[ReferenceRegion] = regions.get ++ regions.get
+              .map(x => ReferenceRegion(x.referenceName.replaceFirst("""^chr""", """"""),
+                x.start,
+                x.end,
+                x.strand))
+
+            // load new dataset or retrieve from cache
+            val data: AlignmentRecordDataset = datasetCache.get(fp) match {
+              case Some(ds) => { // if dataset found in datasetCache
+                ds
+              }
+              case _ => {
+                // load dataset into cache and use use it
+                datasetCache(fp) = sc.loadPartitionedParquetAlignments(fp)
+                datasetCache(fp)
+              }
+            }
+
+            val maybeFiltered = if (finalRegions.nonEmpty) {
+              data.filterByOverlappingRegions(finalRegions)
+            } else data
+
+            // remove unmapped reads
+            maybeFiltered.transformDataset(d => {
+              d.filter(x => (x.readMapped.getOrElse(false)) && x.mappingQuality.getOrElse(0) > 0)
+            })
+
+          } else { // data was not written as partitioned parquet
+            val pred = {
+              val prefixRegions: Iterable[ReferenceRegion] = regions.get.map(r => LazyMaterialization.getReferencePredicate(r)).flatten
+              Some(ResourceUtils.formReferenceRegionPredicate(prefixRegions) && (BooleanColumn("readMapped") === true) && (IntColumn("mappingQuality") > 0))
+            }
+            sc.loadParquetAlignments(fp, optPredicate = pred, optProjection = Some(proj))
           }
+        } else {
+          sc.loadAlignments(fp, optProjection = Some(proj))
+        }
 
-        val proj = Projection(AlignmentRecordField.referenceName, AlignmentRecordField.mappingQuality, AlignmentRecordField.readName,
-          AlignmentRecordField.start, AlignmentRecordField.readMapped, AlignmentRecordField.readGroupId,
-          AlignmentRecordField.end, AlignmentRecordField.sequence, AlignmentRecordField.cigar, AlignmentRecordField.readNegativeStrand,
-          AlignmentRecordField.readPaired, AlignmentRecordField.readGroupSampleId)
-
-        val unpartitionedResult = sc.loadParquetAlignments(fp, optPredicate = pred, optProjection = Some(proj))
-        unpartitionedResult
-      }
-      return alignmentRecordDataset
+      alignmentRecordDataset.rdd.collect()
```

Comment:
Since everything is pulled into RAM on the master node now, is the use case for Mango the same?  Or did a collect or equivalent to collect still happen before?

Reply:
It didn't collect, it did this stupid thing where it would put into distributed memory, then collect later. This is just to simplify the data movement, considering there is no reason to use distributed memory when all you are doing is pulling small amounts of data to the master node at the end of the day. The workload is very different from processing